src/index.ts

```diff
@@ -27,7 +27,13 @@ export default function connect(
   const connectionStatus = new BehaviorSubject<number>(0)
 
   const messages = new Observable<string>(observer => {
-    const socket = websocketFactory(url, protocols)
+
+    let socket;
+    try {
+      socket = websocketFactory(url, protocols);
+    } catch(err) {
+      console.log(err);
```

Comment:
logging the error like this is so arbitrary, it would be better to return a stream that emits an error...

But this whole idea isn't good to me. Isn't it better it just throw the exception so that people know to use `//` or `https://`, I don't think using `http://` is ever a good idea?

Reply:
Even if the library does emit an error here, which is better than logging the error, it's still not something you can ever recover from.